apps/meteor/app/live-location/server/methods/getLiveLocation.ts

```diff
@@ -0,0 +1,77 @@
+import { Messages, Subscriptions } from '@rocket.chat/models';
+import { check } from 'meteor/check';
+import { DDPRateLimiter } from 'meteor/ddp-rate-limiter';
+import { Meteor } from 'meteor/meteor';
+
+import { canAccessRoomIdAsync } from '../../../authorization/server/functions/canAccessRoom';
+
+Meteor.methods({
+	/**
+	 * Get live location data for a message
+	 */
+	async 'liveLocation.get'(rid: string, msgId: string) {
+		check(rid, String);
+		check(msgId, String);
+
+		const uid = Meteor.userId();
+		if (!uid) {
+			throw new Meteor.Error('error-invalid-user', 'Invalid user', { method: 'liveLocation.get' });
+		}
+
+		if (!(await canAccessRoomIdAsync(rid, uid))) {
+			throw new Meteor.Error('error-not-allowed', 'Not allowed', { method: 'liveLocation.get' });
+		}
+
+		const sub = await Subscriptions.findOneByRoomIdAndUserId(rid, uid);
+		if (!sub) {
+			throw new Meteor.Error('error-not-in-room', 'User is not in the room', { method: 'liveLocation.get' });
+		}
+		const msg = await Messages.findOne({
+			_id: msgId,
+			rid,
+			attachments: {
+				$elemMatch: {
+					type: 'live-location',
+				},
+			},
+		});
+
+		if (!msg) {
+			throw new Meteor.Error('error-live-location-not-found', 'Live location not found', {
+				method: 'liveLocation.get',
+			});
+		}
+
+		const attachment = msg.attachments?.find((att: any) => att.type === 'live-location') as any;
```

Comment:
Maybe it should use the message type and not the attachment for defining the type